src/lib/wallet/GoodWalletProvider.js

```diff
@@ -285,7 +287,10 @@ export const GoodWalletProvider = ({ children, disableLoginAndWatch = false }) =
         await goodWallet.setIsPollEvents(false) //stop watching prev chain events
         await goodWallet.init({ network: contractsNetwork }) //reinit wallet
         let web3Provider = new Web3Provider(
-          new PrivateKeyProvider(goodWallet.wallet.eth.accounts.wallet[0].privateKey, goodWallet.wallet._provider.host),
+          new JsonRpcProviderWithSigner(
+            goodWallet.wallet._provider.host,
```

Comment:
```
        let web3Provider = new Web3Provider(
          new JsonRpcProviderWithSigner(
           // this will also use our multiplehttpprovider
            new Web3Provider(goodWallet.wallet.currentProvider),
            goodWallet.wallet.eth.accounts.wallet[0].privateKey,
          ),
        )

Reply:
I believe this is all that was missing